### PR TITLE
Add blosc2_schunk_get_slice_buffer/set_slice_buffer functions

### PR DESCRIPTION
**Pull request by martaiborra**

### CODE REVIEW

File: include/blosc2.h [hunk 1]
```diff
@@ -1804,6 +1804,34 @@ BLOSC_EXPORT int blosc2_schunk_get_chunk(blosc2_schunk *schunk, int64_t nchunk,
 BLOSC_EXPORT int blosc2_schunk_get_lazychunk(blosc2_schunk *schunk, int64_t nchunk, uint8_t **chunk,
                                              bool *needs_free);
 
+/**
+ * @brief Fill buffer with schunk slice.
+ *
+ * @param schunk The super-chunk from where to extract a slice.
+ * @param start The coordinates where the slice will begin (0 indexed).
+ * @param stop The coordinates where the slice will end.
```
Comment (FrancescAlted) [inline include/blosc2.h:1812]: 'Coordinates' is more for multidimensionality.  Suggestions:

"The coordinates where the slice will begin (0 indexed)." -> Index (0-based) where the slice begins.
"The coordinates where the slice will end." -> The first index (0-based) that is not in the selected slice.

File: include/blosc2.h [hunk 2]
```diff
@@ -1804,6 +1804,34 @@ BLOSC_EXPORT int blosc2_schunk_get_chunk(blosc2_schunk *schunk, int64_t nchunk,
 BLOSC_EXPORT int blosc2_schunk_get_lazychunk(blosc2_schunk *schunk, int64_t nchunk, uint8_t **chunk,
                                              bool *needs_free);
 
+/**
+ * @brief Fill buffer with schunk slice.
```
Comment (FrancescAlted) [inline include/blosc2.h:1808]: "Fill buffer with schunk slice." -> "Fill buffer with a schunk slice."

File: include/blosc2.h [hunk 3]
```diff
@@ -1804,6 +1804,34 @@ BLOSC_EXPORT int blosc2_schunk_get_chunk(blosc2_schunk *schunk, int64_t nchunk,
...
+ *
+ * @param schunk The super-chunk from where to extract a slice.
+ * @param start The coordinates where the slice will begin (0 indexed).
+ * @param stop The coordinates where the slice will end.
+ * @param buffer The buffer where the data will be stored.
+ *
+ * @warning You must make sure that you have space enough to store the
```
Comment (FrancescAlted) [inline include/blosc2.h:1815]: "You must make sure that you have space enough to store the" -> You must make sure that you have space enough in @param buffer to store the

File: include/blosc2.h [hunk 4]
```diff
@@ -1804,6 +1804,34 @@ BLOSC_EXPORT int blosc2_schunk_get_chunk(blosc2_schunk *schunk, int64_t nchunk,
...
+ * @return An error code.
+ */
+BLOSC_EXPORT int blosc2_schunk_get_slice_buffer(blosc2_schunk *schunk, int64_t start, int64_t stop, void *buffer);
+
+/**
+ * @brief Set a schunk into a schunk from a buffer.
+ *
+ * @param schunk The super-chunk where to set the slice.
+ * @param start The coordinates where the slice will begin (0 indexed).
+ * @param stop The coordinates where the slice will end.
```
Comment (FrancescAlted) [inline include/blosc2.h:1827]: Ditto about 'coordinates'

**Review by FrancescAlted — CHANGES_REQUESTED** [review]
I have some suggestions.  Also, as this a great feature, it would be nice if you'd add an example exercising get_slice/set_slice functionality.

File: include/blosc2.h (hunk 1 above)
Comment (martaiborra) [reply include/blosc2.h]: Fixed in 1208e4e76f0f085baf1a4e02a8cc276654d40b0f.

File: include/blosc2.h (hunk 2 above)
Comment (martaiborra) [reply include/blosc2.h]: Fixed in 1208e4e76f0f085baf1a4e02a8cc276654d40b0f.

File: include/blosc2.h (hunk 3 above)
Comment (martaiborra) [reply include/blosc2.h]: Fixed in 1208e4e76f0f085baf1a4e02a8cc276654d40b0f.

**Review by FrancescAlted — APPROVED** [review]
LGTM!

**Pull request merged**